### PR TITLE
Add codespell spell checker (makefile, CI)

### PR DESCRIPTION
**Pull request by rockdaboot**

### CODE REVIEW

File: .codespellignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+fo
+raison
```
Comment (dmathieu) [inline .codespellignore:2]: This comes from:
> .//interpreter/php/decode_amd64.c:72:  // The raison d'etre for this function is described in the php8 unwinding doc,

While this is understandable, should we instead use "professionnal" english?

File: .codespellignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+opne
+optimyze
```
Comment (dmathieu) [inline .codespellignore:4]: It seems this name is required for an S3 bucket? Should we plan moving that bucket into the otel org?

File: .codespellignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+fo
+raison
+opne
```
Comment (dmathieu) [inline .codespellignore:3]: I'm not seeing any direct occurence of this. The closest I'm finding is:

> .//zydis/Zydis.c:35676:    ZYDIS_MAKE_SHORTSTRING("loopne"),

Should the issue be fixed instead?

Comment (rockdaboot) [reply .codespellignore:3]: This is from actual code
```
nativeunwind/elfunwindinfo/elfehframe.go
113:    opNE         expressionOpcode = 0x2e
308:            case op == opDeref, op >= opRot && op <= opNE:
```

File: .codespellignore (hunk 1 above)
Comment (rockdaboot) [reply .codespellignore:2]: I thought about exactly this, but then it seemed that `raison d'etre` is a well-known term in English. But now that we are two people already who think about rewording, I'll do it :)

Comment (rockdaboot) [reply .codespellignore]: done

File: .codespellignore (hunk 2 above)
Comment (rockdaboot) [reply .codespellignore:3]: Makes sense to me, but should not be mixed with this PR.

**Review by athre0z — APPROVED** [review]

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+codespell==2.2.6
```
Comment (dmathieu) [inline requirements.txt:1]: As this is now using codespell from apt-get, we shouldn't need all the Python setup anymore?

Comment (rockdaboot) [reply requirements.txt]: Removed

File: .gitignore [hunk 5]
```diff
@@ -6,3 +6,5 @@
 opentelemetry-ebpf-profiler
 ci-kernels
 
+# Installed by `make codespell`
+venv/
```
Comment (dmathieu) [inline .gitignore:10]: This isn't required anymore either, since we use apt-get.

```suggestion
```

**Review by dmathieu — APPROVED** [review]

**Review by christos68k — APPROVED** [review]

**Pull request merged**